boards/nordic/nrf52840dk/src/main.rs

```diff
@@ -64,75 +64,75 @@ pub static mut STACK_MEMORY: [u8; 0x1000] = [0; 0x1000];
 #[no_mangle]
 pub unsafe fn reset_handler() {
     // Loads relocations and clears BSS
-    nrf52::init();
+    nrf52840::init();
 
     // GPIOs
     let gpio_pins = static_init!(
         [&'static dyn kernel::hil::gpio::InterruptValuePin; 13],
         [
             static_init!(
                 kernel::hil::gpio::InterruptValueWrapper,
-                kernel::hil::gpio::InterruptValueWrapper::new(&nrf5x::gpio::PORT[3])
+                kernel::hil::gpio::InterruptValueWrapper::new(&nrf52840::gpio::PORT[3])
             )
             .finalize(), // Bottom right header on DK board
             static_init!(
                 kernel::hil::gpio::InterruptValueWrapper,
-                kernel::hil::gpio::InterruptValueWrapper::new(&nrf5x::gpio::PORT[4])
+                kernel::hil::gpio::InterruptValueWrapper::new(&nrf52840::gpio::PORT[4])
             )
             .finalize(),
             static_init!(
                 kernel::hil::gpio::InterruptValueWrapper,
-                kernel::hil::gpio::InterruptValueWrapper::new(&nrf5x::gpio::PORT[28])
+                kernel::hil::gpio::InterruptValueWrapper::new(&nrf52840::gpio::PORT[28])
             )
             .finalize(),
             static_init!(
                 kernel::hil::gpio::InterruptValueWrapper,
-                kernel::hil::gpio::InterruptValueWrapper::new(&nrf5x::gpio::PORT[29])
+                kernel::hil::gpio::InterruptValueWrapper::new(&nrf52840::gpio::PORT[29])
             )
             .finalize(),
             static_init!(
                 kernel::hil::gpio::InterruptValueWrapper,
-                kernel::hil::gpio::InterruptValueWrapper::new(&nrf5x::gpio::PORT[30])
+                kernel::hil::gpio::InterruptValueWrapper::new(&nrf52840::gpio::PORT[30])
             )
             .finalize(),
             static_init!(
                 kernel::hil::gpio::InterruptValueWrapper,
-                kernel::hil::gpio::InterruptValueWrapper::new(&nrf5x::gpio::PORT[10])
+                kernel::hil::gpio::InterruptValueWrapper::new(&nrf52840::gpio::PORT[10])
```

Comment:
This maps unusable pins, see #1408, especially https://github.com/tock/tock/issues/1408#issuecomment-542312875

Reply:
That's right but re-mapping the GPIOs for nRF52840-dk is independent from this pull-request. Given that there is now quite a bit of refactoring here regarding the dependencies between chips, I'd definitely keep that discussion separate to avoid making this pull-request even more complex.